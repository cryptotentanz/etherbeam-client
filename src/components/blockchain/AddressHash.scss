@use '../../styles/resources/fonts';
@use '../../styles/resources/sizes';

.my-addresshash {
  font-family: fonts.$monospace;
}
