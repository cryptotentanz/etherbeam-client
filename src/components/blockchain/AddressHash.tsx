import React, { FC } from 'react'
import { shortenAddress } from './helpers'
import './AddressHash.scss'

interface AddressHashProps {
  hash: string
  short?: boolean
}

const Hash: FC<AddressHashProps> = ({ hash, short }: AddressHashProps) => {
  return <span className="my-addresshash">{short ? shortenAddress(hash) : hash}</span>
}

export default Hash
