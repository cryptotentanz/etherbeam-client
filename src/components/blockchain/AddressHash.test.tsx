import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import AddressHash from './AddressHash'

describe('<AddressHash />', () => {
  const hash = '0x0000000000000000000000000000000000000111'

  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<AddressHash hash={hash} />)
    })

    it('should render', () => {
      const span = component.find('span')
      expect(span.prop('children')).toBe(hash)
      expect(span.hasClass('my-addresshash')).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Short', () => {
    beforeAll(() => {
      component = shallow(<AddressHash hash={hash} short />)
    })

    it('should render', () => expect(component.find('span').prop('children')).toBe('0x0000...0111'))
  })
})
