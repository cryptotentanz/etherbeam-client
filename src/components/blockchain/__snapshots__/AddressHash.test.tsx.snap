// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressHash /> Default should match snapshot 1`] = `
<span
  class="my-addresshash"
>
  0x0000000000000000000000000000000000000111
</span>
`;
